src/server/ParsingHttpHandler.ts

```diff
@@ -24,25 +27,33 @@ export interface BaseHttpHandlerArgs {
    * Writes out the response of the operation.
    */
   responseWriter: ResponseWriter;
+  /**
+   * Source handler to send the operation to.
+   */
+  source: OperationHttpHandler;
 }
 
 /**
- * Parses requests and sends the resulting Operation to the abstract `handleOperation` function.
+ * Parses requests and sends the resulting Operation to wrapped source.
  * Errors are caught and handled by the Errorhandler.
- * In case the `handleOperation` function returns a result it will be sent to the ResponseWriter.
+ * In case the source returns a result it will be sent to the ResponseWriter.
  */
-export abstract class BaseHttpHandler extends HttpHandler {
-  protected readonly logger = getLoggerFor(this);
+export class ParsingHttpHandler extends HttpHandler {
```

Comment:
To what extent does `Operation[Http]Handler` need to be an `HttpHandler` instead of purely handling (parsed) operations?

Because I wonder if it will be more simple at this point to:
- rename `ParsingHttpHandler` into `OperationHttpHandler` as it performs the HTTP aspect surrounding operations
- rename `OperationHttpHandler` into `OperationHandler` as it performs operations and no longer needs to know about the HTTP envelope

Reply:
I agree that this would be nicer, but this is currently impossible for the following reasons:
 * The `CredentialsExtractor` requires the `HttpRequest` to parse the correct headers.
 * The IDP handler requires the `HttpRequest` to get the OIDC interaction metadata and to determine correct OIDC behavior in case the library needs to handle the request. In that latter case it also requires the `HttpResponse` to write the output to.

Reply:
Should we then pass the request as a separate argument? Because we're now passing `{ operation }`?
Then the operation handlers can choose to just handle operations.